context/src/junit/validator.rs

```diff
@@ -169,8 +170,8 @@ pub fn validate(report: &Report) -> JunitReportValidation {
                         JunitTestCaseValidationIssueSubOptimal::TestCaseOldTimestamp(timestamp),
                     ));
                 } else if time_since_timestamp.num_hours() > i64::from(TIMESTAMP_STALE_HOURS) {
-                    test_case_validation.add_issue(JunitValidationIssue::SubOptimal(
-                        JunitTestCaseValidationIssueSubOptimal::TestCaseStaleTimestamp(timestamp),
+                    report_validation.add_issue(JunitValidationIssue::SubOptimal(
+                        JunitReportValidationIssueSubOptimal::StaleTimestamps,
                     ));
```

Comment:
I'm not sure it makes sense to change collecting `test_case_validation` to `report_validation`. This will make it confusing if later on we want to inspect issues with test cases. Perhaps we can either add logic to derive report validation issues from all child issues? Or we can add to both the test case and report validation, then add logic that dedupes the total issues?

Reply:
> This will make it confusing if later on we want to inspect issues with test cases

Fair point!

> Perhaps we can either add logic to derive report validation issues from all child issues?

Going to head down this path

Reply:
Alrighty - reverted to collecting all individual test case / test suite issues as before, then added a `derive_all_issues()` step afterwards to derive a list of 'all issues to print,' if you will, that includes the report-level issues and handles the deduping. Doing this also eliminated the need for lots of redundant helper functions, subsequently making things cleaner in `cli/src/validate.rs` as well. Lemme know how that looks to you!